app/assets/javascripts/initializers/initializeCommentsPage.js.erb

```diff
@@ -48,7 +48,6 @@ function initializeCommentsPage() {
                     userActionsEl.innerHTML = '<a data-no-instant href="' + userActionsEl.parentNode.dataset.path + '/delete_confirm" class="edit-butt">DELETE</a>\
                                                 <a href="' + userActionsEl.parentNode.dataset.path + '/edit">EDIT</a>'
                     userActionsEl.style.display = 'inline-block';
-                    document.getElementById('button-for-comment-' + allNodes[i].dataset.commentId).classList.add('reacted');
```

Comment:
**Is it safe to remove this?**

This is the bit that was walking through every comment that was self created and rendering it as "liked". This creates inconsistencies with the BE which then causes us to enter weird states when the BE tells us `result: "create"`

Reply:
I agree that this is most likely a problem (always render all self created comments as "liked"). The AJAX response  however (`responseObj`) does include useful data that could possibly be used to put this statement behind an `if` clause.

Adding a debug similar to `console.log("REACTIONS RES:", responseObj)` can help understand the body of the response. Also from the network tab in the devtools you can parse the response from the request and see something like this:

![Screen Shot 2020-09-17 at 10 41 28](https://user-images.githubusercontent.com/6045239/93501267-718b5a00-f8d2-11ea-9897-43342ce06233.png)

You can see that the payload includes the id of the `current_user` up top and all the reaction IDs on the Article. So this new `if` statement to protect the rendering of all self created comments as "liked" should only be true when this `reactions` array  includes the comment with id `allNodes[i].dataset.commentId` and the `user_id` from that reaction matches the `current_user`'s ID.

Reply:
> The AJAX response however (responseObj) does include useful data that could possibly be used to put this statement behind an if clause.

I'll look into this, but I'm pretty sure this is already being handled here:

https://github.com/osfn/forem/blob/11e9e20e4f839cf512b54b688c3e2c9f2d53ddfc/app/assets/javascripts/utilities/buildCommentHTML.js.erb#L102

In our testing, comments that we actually liked still showed the full heart even on refresh.

Reply:
> but I'm pretty sure this is already being handled here:

Oh wait! It's actually looks like it's being handled here 😅:

https://github.com/osfn/forem/blob/11e9e20e4f839cf512b54b688c3e2c9f2d53ddfc/app/assets/javascripts/initializers/initializeCommentsPage.js.erb#L25

Reply:
> Oh wait! It's actually looks like it's being handled here 

You're right, I missed that too. Looks like this may be an unnecessary line after all AFAICT